channeldb/kvdb/etcd/stm.go

```diff
@@ -403,12 +403,23 @@ func (s *stm) Get(key string) ([]byte, error) {
 	// the prefetch set.
 	if getValue, ok := s.prefetch[key]; ok {
 		delete(s.prefetch, key)
-		s.rset[key] = getValue
+
+		// Use the prefetched value only if it is for
+		// an existing key.
+		if getValue.rev != 0 {
```

Comment:
As we assume bucket keys always have a revision of 0 since they're never modified after initial creation?

Reply:
So the idea is that if a key is non-existent then its `ModVersion` is always zero which we use to assert upon commit that certain keys (bucket/value) are non existent.

In this particular case we check if the prefetch set contains the key we're trying to read with a `ModVersion` of 0. The reason such key might be in the prefetch is because upon commit we update the read set values with the transaction response and use that as our prefetch set.